### PR TITLE
Retry op missing for execute+watch

### PR DESCRIPTION
**Pull request by werkt**
Create an execute+watch retriable trigger when an operation is
not found on the server.

Tested with a simulated operation missing, saw an inspired
execute loop retry.

### CODE REVIEW

**Comment by werkt** [comment]
A rebase made a change to the retrier initialization necessary, and upon trying to figure out what the retryService was used for, I decided that this change required a break between the synchronous retrier and the asynchronous one, with the latter requiring a retryService. It made this change bigger, but let RemoteRetrier be only a subclass of the synchronous interface.

**Comment by mhlopko** [comment]
Friendly ping @buchgr

File: src/main/java/com/google/devtools/build/lib/remote/OperationNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.remote;
+
+import java.io.IOException;
+
+/**
+ * An exception to indicate an operation loss.
+ */
```
Comment (buchgr) [inline src/main/java/com/google/devtools/build/lib/remote/OperationNotFoundException.java]: what specifically is an operation loss and why do we want to retry it?

Comment (werkt) [reply src/main/java/com/google/devtools/build/lib/remote/OperationNotFoundException.java]:
```
                        case DOES_NOT_EXIST:
                          // TODO(olaola): either make this retriable, or use a different exception.
                          throw new IOException(
                              String.format(
                                  "Operation %s lost on the remote server.", op.getName()));
```

The name was derived from the above, which it replaces. An operation is lost when a watch request responds with a change state of `DOES_NOT_EXIST`

**Comment by ola-rozenfeld** [comment]
This is now supported in https://github.com/bazelbuild/bazel/commit/f0aa55d81c7a311efef203bdecf7bca99aaded00 for API v2.

**Comment by irengrig** [comment]
Hi All, Bazel sheriff here. Sorry, this looks important, but abandoned. Should any actions be taken? I do not see this commit merged.

**Comment by ola-rozenfeld** [comment]
Yes, this can be abandoned, I submitted a different but equivalent
functionality as part of my API V2 change (see previous comment).

**Comment by irengrig** [comment]
Closing as @ola-rozenfeld suggested.

**Pull request closed without merging**